src/DotPulsar/Internal/ConsumerProcess.cs

```
@@ -84,8 +84,15 @@ protected override void CalculateState()
 
         private async void SetupChannel()
         {
-            var channel = await _factory.Create(CancellationTokenSource.Token).ConfigureAwait(false);
-            await _consumer.SetChannel(channel).ConfigureAwait(false);
+            try
+            {
+                var channel = await _factory.Create(CancellationTokenSource.Token).ConfigureAwait(false);
+                await _consumer.SetChannel(channel).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Setup Channel failed for Consumer \n {ex.ToString()}");
```

Let's remove the output to the console here. We don't know if there even is a console and the exception thrown has already been through the exception handling pipeline (and outputted if that is what the user wanted).
So, we can have an empty catch.
```csharp
catch
{
    // ignored
}
```
The same goes for the ProducerProcess and we also need this for the ReaderProcess.

---

Yeah, good point. Changed it.